libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Settings/CoreBotOptions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Builder.Integration.Runtime.Settings
+{
+    /// <summary>
+    /// Defines options to be supplied to <see cref="CoreBot"/>.
+    /// </summary>
+    internal class CoreBotOptions
+    {
+        /// <summary>
+        /// Gets or sets the default locale to be utilized by the bot. Defaults to 'en-US'.
+        /// </summary>
+        /// <value>
+        /// The default locale to be utilized by the bot. Defaults to 'en-US'.
+        /// </value>
+        public string DefaultLocale { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource identifier of the dialog to serve as the root dialog of the bot.
+        /// </summary>
+        /// <value>
+        /// The resource identifier of the dialog to serve as the root dialog of the bot.
+        /// </value>
+        public string RootDialog { get; set; }
```

Comment:
The root dialog can be well-known - like main in UNIX/C

Reply:
It is well known and passed to the core bot in this variable.

Reply:
This comes from outside because it is computed based on analyzing the folder structure. There is a well known locationm for the root dialog but this class does not know which file is in that location. We cannot have a fixed name for the file yet because composer makes a dynamic name for the root dialog based on the bot name. We can get to a single name dialog but need a few iterations for downstream dependency. Note that however this root dialog is computed inthe configuration extensions and not loaded from config which is a step in the direction you are proposing.